### PR TITLE
qemu.test: fix random time issue in migration_multi_host_with_reboot

### PR DESCRIPTION
**Pull request by XueqiangWei**
Signed-off-by: Xueqiang Wei <xuwei@redhat.com>
ID: 1410700

### CODE REVIEW

**Comment by XueqiangWei** [comment]
06:36:46 INFO | 			GOOD	virt.case4.remote_nfs.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.multi_host.migrate_multi_host.with_reboot.vhost_src_yes_dst_yes.mig_online.tcp	virt.case4.remote_nfs.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.multi_host.migrate_multi_host.with_reboot.vhost_src_yes_dst_yes.mig_online.tcp	timestamp=1483684608	localtime=Jan 06 06:36:48	completed successfully
06:36:46 INFO | 		END GOOD	virt.case4.remote_nfs.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.multi_host.migrate_multi_host.with_reboot.vhost_src_yes_dst_yes.mig_online.tcp	virt.case4.remote_nfs.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.multi_host.migrate_multi_host.with_reboot.vhost_src_yes_dst_yes.mig_online.tcp	timestamp=1483684608	localtime=Jan 06 06:36:48	
06:36:47 INFO | 	END GOOD	----	----	timestamp=1483684608	localtime=Jan 06 06:36:48	
06:36:49 INFO | Client complete
06:37:01 INFO | 			GOOD	virt.case4.remote_nfs.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.multi_host.migrate_multi_host.with_reboot.vhost_src_yes_dst_yes.mig_online.tcp	virt.case4.remote_nfs.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.multi_host.migrate_multi_host.with_reboot.vhost_src_yes_dst_yes.mig_online.tcp	timestamp=1483684622	localtime=Jan 06 06:37:02	completed successfully
06:37:01 INFO | 		END GOOD	virt.case4.remote_nfs.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.multi_host.migrate_multi_host.with_reboot.vhost_src_yes_dst_yes.mig_online.tcp	virt.case4.remote_nfs.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.multi_host.migrate_multi_host.with_reboot.vhost_src_yes_dst_yes.mig_online.tcp	timestamp=1483684622	localtime=Jan 06 06:37:02	
06:37:01 INFO | 	END GOOD	----	----	timestamp=1483684623	localtime=Jan 06 06:37:03

**Comment by CongLi** [comment]
@Zhengtong 

Could you please help test it?

**Comment by CongLi** [comment]
@Zhengtong 

Could you please help review it?

**Comment by CongLi** [comment]
LGTM,

Acked-by: Cong Li <coli@redhat.com>

File: multi_host_migration/tests/migration_multi_host_with_reboot.py [hunk 1]
```diff
@@ -84,7 +84,7 @@ def migration_scenario(self, worker=None):
                 min_t = int(params.get("min_random_timeout"))
                 max_t = int(params.get("max_random_timeout"))
```
Comment (xutian) [inline multi_host_migration/tests/migration_multi_host_with_reboot.py]: see a bug here.  it should be params["max/min_random_timeout"], covert  None object to int will raise TypeError here

File: multi_host_migration/tests/cfg/multi_host.cfg [hunk 2]
```diff
@@ -75,6 +75,7 @@
                     paused_after_start_vm = yes
                     preprocess_env = yes
                     enable_random_timeout = yes
+                    start_migration_timeout = random
```
Comment (xutian) [inline multi_host_migration/tests/cfg/multi_host.cfg]: not not use param 'enable_radom_timeout' directly.

File: multi_host_migration/tests/migration_multi_host_with_reboot.py (hunk 1 above)
Comment (XueqiangWei) [inline multi_host_migration/tests/migration_multi_host_with_reboot.py]: I will add default value. thx

File: multi_host_migration/tests/cfg/multi_host.cfg (hunk 2 above)
Comment (XueqiangWei) [inline multi_host_migration/tests/cfg/multi_host.cfg]: just see "start_migration_timeout" used in code, so add in cfg file. I will update them. Many thanks

**Comment by XueqiangWei** [comment]
08:30:08 INFO | 			GOOD	virt.case2.remote_nfs.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.multi_host.migrate_multi_host.with_reboot.vhost_src_yes_dst_yes.mig_online.tcp	virt.case2.remote_nfs.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.multi_host.migrate_multi_host.with_reboot.vhost_src_yes_dst_yes.mig_online.tcp	timestamp=1483691410	localtime=Jan 06 08:30:10	completed successfully
08:30:08 INFO | 		END GOOD	virt.case2.remote_nfs.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.multi_host.migrate_multi_host.with_reboot.vhost_src_yes_dst_yes.mig_online.tcp	virt.case2.remote_nfs.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.multi_host.migrate_multi_host.with_reboot.vhost_src_yes_dst_yes.mig_online.tcp	timestamp=1483691410	localtime=Jan 06 08:30:10	
08:30:08 INFO | 	END GOOD	----	----	timestamp=1483691410	localtime=Jan 06 08:30:10	
08:30:10 INFO | Client complete
08:30:41 INFO | 			GOOD	virt.case2.remote_nfs.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.multi_host.migrate_multi_host.with_reboot.vhost_src_yes_dst_yes.mig_online.tcp	virt.case2.remote_nfs.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.multi_host.migrate_multi_host.with_reboot.vhost_src_yes_dst_yes.mig_online.tcp	timestamp=1483691443	localtime=Jan 06 08:30:43	completed successfully
08:30:41 INFO | 		END GOOD	virt.case2.remote_nfs.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.multi_host.migrate_multi_host.with_reboot.vhost_src_yes_dst_yes.mig_online.tcp	virt.case2.remote_nfs.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.multi_host.migrate_multi_host.with_reboot.vhost_src_yes_dst_yes.mig_online.tcp	timestamp=1483691443	localtime=Jan 06 08:30:43	
08:30:41 INFO | 	END GOOD	----	----	timestamp=1483691443	localtime=Jan 06 08:30:43

File: multi_host_migration/tests/migration_multi_host_with_reboot.py [hunk 3]
```diff
@@ -80,11 +80,11 @@ def migration_scenario(self, worker=None):
...
 
-            if params.get("start_migration_timeout") == "random":
-                min_t = int(params.get("min_random_timeout"))
-                max_t = int(params.get("max_random_timeout"))
+            if params.get("enable_random_timeout") == "yes":
+                min_t = int(params.get("min_random_timeout", 1))
+                max_t = int(params.get("max_random_timeout", 5))
                 self.random_timeout = random.randint(min_t, max_t)
-                params["start_migration_timeout"] = 0
+                params["start_migration_timeout"] = self.random_timeout
```
Comment (suqinhuang) [inline multi_host_migration/tests/migration_multi_host_with_reboot.py]: it better to update                 
params["start_migration_timeout"] = random_timeout in migration_multi_host.py
to enable_random_timeout as well

too many params make  puzzled

**Comment by PyLearner** [comment]
LGTM

**Comment by Zhengtong** [comment]
Logic is OK to me.
But I think It may be better to add a log info  to trace the "self.random_timeout" value for each test run, if necessary.
Like this 
params["start_migration_timeout"] = self.random_timeout
**error.context("Start migration after  %d seconds" % self.random_timeout, logging.info)**

**Comment by XueqiangWei** [comment]
07:34:12 INFO | 			GOOD	virt.case3.remote_nfs.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.multi_host.migrate_multi_host.with_reboot.vhost_src_yes_dst_yes.mig_online.tcp	virt.case3.remote_nfs.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.multi_host.migrate_multi_host.with_reboot.vhost_src_yes_dst_yes.mig_online.tcp	timestamp=1484033654	localtime=Jan 10 07:34:14	completed successfully
07:34:12 INFO | 		END GOOD	virt.case3.remote_nfs.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.multi_host.migrate_multi_host.with_reboot.vhost_src_yes_dst_yes.mig_online.tcp	virt.case3.remote_nfs.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.multi_host.migrate_multi_host.with_reboot.vhost_src_yes_dst_yes.mig_online.tcp	timestamp=1484033654	localtime=Jan 10 07:34:14	
07:34:12 INFO | 	END GOOD	----	----	timestamp=1484033655	localtime=Jan 10 07:34:15	
07:34:14 INFO | Client complete
07:34:36 INFO | 			GOOD	virt.case3.remote_nfs.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.multi_host.migrate_multi_host.with_reboot.vhost_src_yes_dst_yes.mig_online.tcp	virt.case3.remote_nfs.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.multi_host.migrate_multi_host.with_reboot.vhost_src_yes_dst_yes.mig_online.tcp	timestamp=1484033678	localtime=Jan 10 07:34:38	completed successfully
07:34:36 INFO | 		END GOOD	virt.case3.remote_nfs.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.multi_host.migrate_multi_host.with_reboot.vhost_src_yes_dst_yes.mig_online.tcp	virt.case3.remote_nfs.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.multi_host.migrate_multi_host.with_reboot.vhost_src_yes_dst_yes.mig_online.tcp	timestamp=1484033678	localtime=Jan 10 07:34:38	
07:34:36 INFO | 	END GOOD	----	----	timestamp=1484033678	localtime=Jan 10 07:34:38

**Comment by Zhengtong** [comment]
Acked
Zhengtong(zhengtli@redhat.com)

**Pull request merged**